src/main/frontend/handler/editor/keyboards.cljs

```diff
@@ -14,8 +14,14 @@
      :on-hide
      (fn [_state e event]
        (let [target (.-target e)]
-         (if (d/has-class? target "bottom-action") ;; FIXME: not particular case
+         (cond
+           (state/get-editor-action)
```

Comment:
I'm also unsure if this will negatively impact any uses of `editor/box`. We could just do a set check for allowed actions that trigger this behavior if we'd like to be safe and more explicit of what this code does

Reply:
Do you know how I can set up `editor/box` somewhere so I can test with it? I'm still not quite sure what it is/how to create one.

Reply:
`frontend.components.editor/box` is the underlying component for all the autocompletion menus so you've already tested it. I just mentioned it because any component that uses `editor/box` gets this escape behavior

Reply:
Ah! I see; I should have pulled up the react devtools component tree sooner.

Sure, I'll add a whitelist for actions that escape should cancel.

Reply:
Yup, you were correct to suggest that. In fact, the `:input` case needed some special handling since it opens up a separate dialog that removes focus from the textarea.

I also omitted :zotero, since if you don't have it configured it opens a whole new page, and I don't know what happens if you do have it configured.

Reply:
After making these recent changes, I found that at least for `:input`, deferring the actual handling of the event to the input component worked well (and then in editor/keyboard.cljs just doing nothing if one of those other things would fire).

I'm not sure it would be better to do it this way for the other cases as well, since it would mean adding this "clear the current action" line to multiple other places instead of just this one (at least in the autocomplete modal and date-picker), and it would require keeping multiple places in sync with editor/keyboard about which actions should have their Escape deferred to some other component's key handler. If you think it would be better for modularity, I can try it out, but I expect it to be uglier than the way this currently is.

Best would be if there was a way to automatically detect if something else was going to handle escape, and not do anything in editor/keyboard if so, but I don't know how to do that reliably. And doing it just with listener creation order (which is currently in the wrong order anyway) feels fragile. Maybe adding a new state variable to track that could be good, but that would require a bunch of new lines to set/unset it.